tensorflow/contrib/learn/python/learn/learn_io/data_feeder.py

```diff
@@ -507,10 +507,13 @@ def _feed_dict_fn():
           inp[i, :] = six.next(self._x)
         except StopIteration:
           self.stopped = True
-          inp = inp[:i, :]
-          if self._y is not None:
-            out = out[:i]
-          break
+          if i==0:
+            raise StopIteration
```

Comment:
I meant, omit the `StopIteration`. `raise StopIteration` will create a new `StopIteration` instance with a stack trace to line 511, while the original `StopIteration is discarded. If you instead simply write `if i == 0: raise`, you will reuse the original exception, which contains the full stack trace.

It doesn't matter much in this case (not much useful information in a `StopIteration` anyway, and it'll be caught shortly by ourselves), but it's better to preserve it anyway.